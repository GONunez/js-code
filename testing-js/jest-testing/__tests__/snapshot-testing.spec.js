const { getFlyingSuperHeros } = require("../classes/super-heros");

//painful
test("manually validate the data", () => {
  const flyingHeros = getFlyingSuperHeros();
  console.log(flyingHeros);
  expect(flyingHeros).toEqual([
    { name: "Dynaguy", powers: ["disintegration ray", "fly"] },
  ]);
});
//using snapshot files
test("validate using a snapshot", function () {
  const flyingHeros = getFlyingSuperHeros();
  console.log(flyingHeros);
  expect(flyingHeros).toMatchSnapshot();
});

//using inline snapshots
test("uses inline snapshot", () => {
  const flyingHeros = getFlyingSuperHeros();
  console.log(flyingHeros);
  expect(flyingHeros).toMatchInlineSnapshot(`
    Array [
      Object {
        "name": "Dynaguy",
        "powers": Array [
          "disintegration ray",
          "fly",
        ],
      },
    ]
  `);
});
