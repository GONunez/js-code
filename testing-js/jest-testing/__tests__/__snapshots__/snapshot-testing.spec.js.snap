// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validate using a snapshot 1`] = `
Array [
  Object {
    "name": "Dynaguy",
    "powers": Array [
      "disintegration ray",
      "fly",
    ],
  },
]
`;
